packages/design-system/src/components/NumericInputField/NumericInput.js

```diff
@@ -146,25 +146,28 @@ class NumericInput extends React.PureComponent {
       className
     );
 
+
     return (
       <div className={mergedClassNames} style={this.getComponentStyles()}>
         <input
-          type="text"
+          type="number"
           {...restProps}
           value={value}
           disabled={disabled}
           onChange={this.handleChange}
           onFocus={this.handleFocus}
           onBlur={this.handleBlur}
+          min={min}
+          max={max}
         />
         {!noControls && (
           <React.Fragment>
             <button
               tabIndex="-1"
               disabled={
                 disabled ||
-                (this.props.max &&
-                  parseInt(this.props.value, 10) === this.props.max)
+                (max &&
+                  parseInt(value, 10) === max)
```

Comment:
I can see that the logic didn't change here, but is it correct that we these buttons don't treat `0` as either a valid min or max value?

Reply:
@kamilmateusiak could you comment? I don't know what did you have in mind :D

Reply:
@opif means that `max && parseInt(value, 10) === max` will turn out to be falsy here when the max is 0. You could check is `max !== undefined` (let's not be to strict here, there is no need to check for null and etc.)

Reply:
there is a couple of checks for `min` and `max` props in the code so you could add a helper `hasMax => this.props.max !== undefined`

Reply:
@kamilmateusiak  please look at 4f781f0dc390a57e0b0c8f5de6e8171a7e5c3448